spec/features/01_user_auth_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+feature 'user is able to' do
+
+    let!(:test_user) {User.create!(email: "test_user@mail.com", password: "password", username: "test_user_2", first_name: "test")}
+    
+    scenario 'click sign up' do
+        visit "/"
+
+        click_link "Sign Up"
+        checks = ["Username", "Password", "Email"]
+
+        checks.each do |check|
+            expect(page).to have_content(check)
+        end
+    end
+
+    scenario 'is able to sign up' do
+        visit new_user_registration_path
+
+        fill_in 'Username' , with: "new_user"
+        fill_in 'Password' , with: "password"
+        fill_in 'Password confirmation' , with: "password"
+        fill_in 'Email' , with: "test@mail.com"
+        fill_in 'First name' , with: "New"
+        fill_in 'Last name' , with: "User"
+
+        click_button "Sign up"
+
+        expect(page).to have_content("signed up successfully")
+    end
+
+    scenario 'is able to sign in and out' do
```

Comment:
Let's break this spec into 3 smaller scoped features:
1. User can sign up
2. User can sign in
3. User can sign out